include/seqan3/alignment/matrix/detail/alignment_score_matrix_one_column_banded.hpp

```diff
@@ -0,0 +1,200 @@
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    //!\brief Defaulted.
+    constexpr alignment_score_matrix_one_column_banded() = default;
+    //!\brief Defaulted.
+    constexpr alignment_score_matrix_one_column_banded(alignment_score_matrix_one_column_banded const &) = default;
+    //!\brief Defaulted.
+    constexpr alignment_score_matrix_one_column_banded(alignment_score_matrix_one_column_banded &&) = default;
+    //!\brief Defaulted.
+    constexpr alignment_score_matrix_one_column_banded &
+    operator=(alignment_score_matrix_one_column_banded const &) = default;
+    //!\brief Defaulted.
+    constexpr alignment_score_matrix_one_column_banded &
+    operator=(alignment_score_matrix_one_column_banded &&) = default;
+    //!\brief Defaulted.
+    ~alignment_score_matrix_one_column_banded() = default;
+
+    /*!\brief Construction from two ranges and a band.
+     * \tparam first_sequence_t  The first range type; must model std::ranges::ForwardRange.
+     * \tparam second_sequence_t The second range type; must model std::ranges::ForwardRange.
+     * \param[in] first         The first range.
+     * \param[in] second        The second range.
+     * \param[in] band          The band in which to calculate the alignment.
+     * \param[in] initial_value The value to initialise the matrix with. Default initialised if not specified.
+     *
+     * \details
+     *
+     * Does only obtain the sizes of the passed ranges in order to allocate the score matrix. Only allocates
+     * one column of the size of the band.
+     */
+    template <std::ranges::ForwardRange first_sequence_t,
+              std::ranges::ForwardRange second_sequence_t>
+    constexpr alignment_score_matrix_one_column_banded(first_sequence_t && first,
+                                                       second_sequence_t && second,
+                                                       static_band const & band,
+                                                       score_t const initial_value = score_t{})
+    {
+        matrix_base_t::cols = static_cast<size_t>(std::ranges::distance(first) + 1);
+        matrix_base_t::rows = static_cast<size_t>(std::ranges::distance(second) + 1);
+
+        band_col_index = std::min<int32_t>(std::max<int32_t>(band.upper_bound, 0), matrix_base_t::cols - 1);
+        band_row_index = std::min<int32_t>(std::abs(std::min<int32_t>(band.lower_bound, 0)), matrix_base_t::rows - 1);
+        band_size = band_col_index + band_row_index + 1;
+
+        // Reserve one more cell to deal with last cell in the banded column which needs only the diagonal and up cell.
+        matrix_base_t::pool.resize(band_size + 1, element_type{initial_value, initial_value});
+    }
+    //!\}
+
+private:
+
+    //!\brief The column index where the upper bound of the band passes through.
+    int32_t band_col_index{};
+    //!\brief The row index where the lower bound of the band passes through.
+    int32_t band_row_index{};
```

Comment:
The description is not so good as one can define the band in a way that the lower bound doesn't pass though any row because it is larger than the main diagonal... you probably expect a negative row index then.

Reply:
No, the point is that except for very specific reasons, e.g. when doing bit operations, one should always prefer the signed types. This is however in many cases already messed up because the size type for all containers etc are unsigned. But in this specific case I do a lot of arithmetic operations and it helps to have them as signed integers, especially since I am getting negative values on purpose for some computations.

Reply:
Good, then better leave it signed.